zerver/models.py

```diff
@@ -1110,11 +1110,19 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
     """
     EMBEDDED_BOT = 4
 
+    """
+    Administrator Bot" bot type, which only admins can create or own,
+    and otherwise works like a Generic Bot but has administrator permissions
+    (is_realm_admin=True) and thus can do things like create users etc.
+    """
+    ADMINISTRATOR_BOT = 5
```

Comment:
Will it be possible to have an embedded bot/outgoing webhook bot works as an administrator in the future? The current administrator bot idea lgtm though.